tests/devices/test_default_clifford.py

```diff
@@ -94,38 +93,276 @@ def circuit_fn():
+        qml.s_prod(3.0, qml.PauliX(0)),
+        qml.sum(qml.PauliX(0), qml.s_prod(2.0, qml.PauliY(1))),
+        qml.prod(qml.sum(qml.PauliZ(0), qml.PauliY(1)), qml.s_prod(3, qml.PauliX(2))),
+        qml.Hermitian(qml.sum(qml.PauliX(0), qml.PauliY(1)).matrix(), [0, 1]),
+        qml.Projector([1, 0], [0, 1]),
+    ],
+)
+def test_meas_expval(shots, ops):
+    """Test that expectation value measurements with `default.clifford` is possible
+    and agrees with `default.qubit`."""
+    dev_c = qml.device("default.clifford", shots=shots, seed=24)
+    dev_q = qml.device("default.qubit")
+
+    def circuit_fn():
+        circuit_1()
+        return qml.expval(ops)
+
+    qnode_clfrd = qml.QNode(circuit_fn, dev_c)
+    qnode_qubit = qml.QNode(circuit_fn, dev_q)
+
+    assert np.allclose(qnode_clfrd(), qnode_qubit(), atol=1e-2 if shots else 1e-8)
+
+
+@pytest.mark.parametrize("shots", [None, int(2e6)])
+@pytest.mark.parametrize(
+    "ops",
+    [
+        qml.PauliZ(0),
+        qml.PauliZ(0) @ qml.PauliY(1),
+        qml.sum(qml.PauliZ(0), qml.PauliY(1)),
+        qml.prod(qml.PauliZ(0), qml.PauliY(1)),
+        qml.s_prod(3.0, qml.PauliX(0)),
+        qml.sum(qml.PauliZ(0), qml.s_prod(2.0, qml.PauliY(1))),
+    ],
+)
+def test_meas_var(shots, ops):
+    """Test that variance measurements with `default.clifford` is possible
+    and agrees with `default.qubit`."""
+    dev_c = qml.device("default.clifford", shots=shots)
+    dev_q = qml.device("default.qubit")
+
+    def circuit_fn():
+        circuit_1()
+        return qml.var(ops)
+
+    qnode_clfrd = qml.QNode(circuit_fn, dev_c)
+    qnode_qubit = qml.QNode(circuit_fn, dev_q)
+
+    assert np.allclose(qnode_clfrd(), qnode_qubit(), atol=1e-2 if shots else 1e-8)
+
+
+@pytest.mark.parametrize("shots", [1024, 8192, 16384])
+def test_meas_samples(shots):
+    """Test if samples are returned with shots given in the clifford device."""
+
+    @qml.qnode(qml.device("default.clifford", shots=shots))
+    def circuit_fn():
+        qml.BasisState(np.array([1, 1]), wires=range(2))
+        return [
+            qml.sample(wires=[1]),
+            qml.sample(qml.PauliZ(0)),
+            qml.sample(qml.PauliX(0) @ qml.PauliY(1)),
+        ]
+
+    samples = circuit_fn()
+    assert len(samples) == 3
+    assert qml.math.shape(samples[0]) == (shots, 1)
+    assert qml.math.shape(samples[1]) == (shots, 1)
+    assert qml.math.shape(samples[2]) == (shots, 2)
+
+
+@pytest.mark.parametrize("tableau", [True, False])
+@pytest.mark.parametrize("shots", [None, 8192])
+@pytest.mark.parametrize(
+    "ops",
+    [
+        None,
+        qml.PauliY(0),
+        qml.PauliX(0) @ qml.PauliY(1),
+        qml.PauliX(0) @ qml.PauliY(1) @ qml.PauliZ(2),
+    ],
+)
+def test_meas_probs(tableau, shots, ops):
+    """Test if probabilities are returned in the clifford device."""
+
+    dev_c = qml.device("default.clifford", tableau=tableau, shots=shots, seed=24)
+    dev_q = qml.device("default.qubit")
```

Comment:
No need for shots here?

Reply:
I am testing the sampling results from `default.clifford` against the analytic solution from `default.qubit`, so no shots are required for the latter.